### PR TITLE
V1.3

### PR DESCRIPTION
**Pull request by xrotwang**
Here's what could become CLDF 1.3. Please have a look @jcgood @fmatter @Glottotopia

### CODE REVIEW

**Comment by d97hah** [comment]
Looks good to me

**Review by LinguList — APPROVED** [review]
Looks nice to me, many thanks!

**Comment by jcgood** [comment]
Looks good to me, too, though I'm still new to reviewing these sorts of things. Thanks!

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,18 @@
...
+
+## [1.3] - ...
+
+### Added
+
+- New component `ParameterNetwork` and associated properties.
+- New module `TextCorpus` and associated properties.
+- New property `treeReference` to relate data to objects in `TreeTable`
+- New language property `parentLanguageGlottocode`.
+- New language property `speakerArea`.
```
Comment (SimonGreenhill) [inline CHANGELOG.md:16]: Is there a meaningful difference in the camel casing here e.g. *T*extCorpus vs. *t*reeReference?

File: README.md [hunk 2]
```diff
@@ -80,15 +80,32 @@ i.e. the CSV dialect specified as:
 }
 ```
 
-Some of the effects of this metadata-free conformance are:
-- The first line of each file must contain the comma-separated list of column names.
-- No comment lines are allowed.
+For the single CSV file of a metadata-free CLDF dataset
```
Comment (SimonGreenhill) [inline README.md:83]: For a single CSV file to be a CLDF-compliant dataset without metadata

File: README.md [hunk 3]
```diff
@@ -80,15 +80,32 @@ i.e. the CSV dialect specified as:
...
+```
 
 
 ### Extended conformance
 
-A dataset is CLDF conformant if it uses a custom metadata file, derived from the default profile for the appropriate 
-module. The metadata MUST specify a `dc:conformsTo` property with one of the CLDF module URIs as value.
+A dataset is CLDF conformant if 
+- it contains a metadata file, derived from the default profile for the appropriate module,
+- at least the minimal set of components (i.e. CSV data files) specified for the module.
```
Comment (SimonGreenhill) [inline README.md:101]: "it contains the minimal set of components at least (i.e. ...)"

File: README.md [hunk 4]
```diff
@@ -376,6 +432,22 @@ encoded BibTeX file (with the citation keys serving as local Source IDs). The fi
...
+
+CLDF datasets may contain large data or reference files which may be inconvenient (e.g. because the size exceeds GitHub's
+100MB filesize limit). In such cases, the dataset creator may compress individual files using the
+[ZIP format](https://en.wikipedia.org/wiki/ZIP_(file_format)) (which works really well on CSV and BibTeX files).
+The resulting ZIP archive MUST contain the zipped file as single member and MUST be named after the original
+file, adding `.zip` as filename extension. The filename references in the metadata MUST be kept unchanged.
+
+CLDF processing software MAY implement zip-file discovery, i.e. if a filename referenced in the metadata cannot
+be found, but a file `filename.zip` is found, processing MUST proceed with the unzipped content of `filename.zip`.
+
```
Comment (SimonGreenhill) [inline README.md:445]: should this be filename.ext.zip? e.g. we have `posterior.trees.zip` in phlorest and `sources.bib.zip` in e.g. glottolog

File: components/README.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+# CLDF Components
+
+CLDF components are basically tables which model (cross-linguistically meaningful) datatypes.
```
Comment (SimonGreenhill) [inline components/README.md:3]: delete 'basically'

File: components/examples/description.md [hunk 6]
```diff
@@ -28,6 +28,20 @@ In such cases it is recommended to consistently use U+2026 - i.e. `…`, the [Un
...
+[grammaticality judgement](https://github.com/cysouw/pandoc-ling?tab=readme-ov-file#the-basic-structure-of-a-linguistic-example).
+
+In CLDF, ungrammatical examples MUST
+- have a non-empty [`grammaticalityJudgement`](http://cldf.clld.org/v1.0/terms.rdf#grammaticalityJudgement) value, namely
+  the typographical marker to be used for the example,
+- link (via `languageReference`) to special item(s) in `LanguageTable` with an empty `Glottocode` to
+  prevent data aggregators from inadvertently assigning such an example to a proper language
+  (if they fail to honour `grammaticalityJudgement`).
+
+
```
Comment (SimonGreenhill) [inline components/examples/description.md:44]: this section is also in `README.md` - DRY?

File: components/parameternetworks/README.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+for this, a format that allows discovery of this intent should be prefered. E.g. 
+[GraphML](http://graphml.graphdrawing.org/) - being namespaced XML - can easily be detected from a 
+`mediaType` of `text/xml`, and the namespace declaration in the XML, while GraphViz' DOT files do not 
+have such a transparent discovery mechanism (see https://forum.graphviz.org/t/mime-type-for-downloading-dot-files/877).
+
+A `parameterEdgeReference` property can be used to link rows in CLDF tables to particular edges in
+a `ParameterNetwork`. E.g. a custom table listing individual colexifications, i.e. pairs of word counterparts
+for different meanings in a `Wordlist`, could link each colexification to the edge in the concept network
+that is corroborated by the colexfication.
+
```
Comment (SimonGreenhill) [inline components/parameternetworks/README.md:26]: Note that this format would also work for a tree (a tree is an acyclic, bifurcating, network)

File: extensions/sql.md [hunk 8]
```diff
@@ -23,6 +23,16 @@ MUST be used for database objects, i.e.
 - non-CLDF columns keep their name.
 
 
+## Data types
+
+As much s possible, [CSVW datatypes](https://www.w3.org/TR/2015/REC-tabular-metadata-20151217/#built-in-datatypes)
```
Comment (SimonGreenhill) [inline extensions/sql.md:28]: "much as"

File: modules/README.md [hunk 9]
```diff
@@ -0,0 +1,12 @@
+# CLDF Modules
+
+CLDF Modules are used to model complex cross-linguistic datatypes, which typically comprise
+multiple [components](../components) - and may give the contained components particular semantics.
+E.g. an `ExampleTable` in a `StructureDataset` typically holds examples, illustrating particular
+feature codings in the dataset, whereas an `ExampleTable` in a `TextCorpus` holds the annotated
+text of the corpus.
+
+CLDF consumers can use the additional semantics conveyed through the module choice to inform
+processing. E.g. a line of glossed text may be formatted as IGT example, including number, source
```
Comment (SimonGreenhill) [inline modules/README.md:10]: `E.g.` -> "For example, a"

**Review by SimonGreenhill — COMMENTED** [review]
looks good @xrotwang just some minor comments from me, mainly wording suggestions.

File: README.md [hunk 10]
```diff
@@ -30,11 +30,11 @@ interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
 ## Conformance Levels
 
 CLDF is based on W3C's suite of specifications for [CSV on the Web](https://www.w3.org/TR/tabular-data-primer/), 
-or short CSVW.
+or short CSVW. Thus, cross-linguistic data in CLDF is modeled as inter-related tabular data.
```
Comment (chrzyki) [inline README.md:33]: Just 'interrelated'.

File: CHANGELOG.md (hunk 1 above)
Comment (xrotwang) [reply CHANGELOG.md:16]: Yes, module and component names start with uppercase, property names start with lowercase.

File: README.md [hunk 11]
```diff
@@ -80,15 +80,32 @@ i.e. the CSV dialect specified as:
...
-module. The metadata MUST specify a `dc:conformsTo` property with one of the CLDF module URIs as value.
+A dataset is CLDF conformant if 
+- it contains a metadata file, derived from the default profile for the appropriate module,
+- at least the minimal set of components (i.e. CSV data files) specified for the module. 
+
+The metadata MUST specify a `dc:conformsTo` property with one of the CLDF module URIs as value.
+
+> [!TIP]
+> Thus, a minimal extended CLDF [StructureDataset](modules/StructureDataset) will consist of
+> - a JSON file containing the metadata (whose name can be freely chosen),
```
Comment (chrzyki) [inline README.md:107]: I'd replace all of the 'whose' referring to file names etc. with something like "a JSON file containing the metadata (with a freely chosen name)".

File: README.md [hunk 12]
```diff
@@ -80,15 +80,32 @@ i.e. the CSV dialect specified as:
...
+> - a CSV file containing the dataset's [ValueTable](components/values) (whose name is specified in the metadata).
```
Comment (chrzyki) [inline README.md:108]: See above.

File: README.md [hunk 13]
```diff
@@ -114,35 +131,74 @@ Thus, using extended conformance via metadata, a dataset may:
...
-  when a term was introduced using `dc:hasVersion` properties per term.
-- For better human readability the [CLDF Ontology](http://cldf.clld.org/v1.0/terms.rdf) should
-be visited with a browser capable of rendering XSLT - such as Firefox.
+> [!CAUTION]
+> In an ill-advised attempt to version the ontology, `v1.0` has been baked into the term
+> URIs. While this may be a good idea in case of incompatible changes (e.g. if the semantics
+> of a term changed), it presents an obstacle for interoperability in case of backwards-compatible
+> changes. So starting with CLDF 1.1, we will keep `http://cldf.clld.org/v1.0/terms.rdf`
+> as namespace for **all** versions of the 1.x series, and specify the particular version
+> when a term was introduced using `dc:hasVersion` properties per term.
```
Comment (chrzyki) [inline README.md:197]: No real comment for changes, just wanted to mention that I think this is a very important comment (and helpful clarification) for people reading the specification, particularly when they encounter the first `dc:conformsTo` in the minimal example.

File: README.md (hunk 4 above)
Comment (xrotwang) [reply README.md:445]: I'll rewrite to `filename_with_extension.zip` for maximum transparency. For me, "filename" already includes the extension - while "stem" is the filename with the - first level of - extension(s) split off. But I guess that intuition may be different in the windows world.

File: README.md [hunk 14]
```diff
@@ -376,6 +432,22 @@ encoded BibTeX file (with the citation keys serving as local Source IDs). The fi
 - or specified as path relative to the metadata file given for top-level common property `dc:source` in the dataset's metadata.
 
 
+### Compressed data or reference files
+
+CLDF datasets may contain large data or reference files which may be inconvenient (e.g. because the size exceeds GitHub's
+100MB filesize limit). In such cases, the dataset creator may compress individual files using the
+[ZIP format](https://en.wikipedia.org/wiki/ZIP_(file_format)) (which works really well on CSV and BibTeX files).
+The resulting ZIP archive MUST contain the zipped file as single member and MUST be named after the original
```
Comment (chrzyki) [inline README.md:440]: "MUST contain only the zipped file and nothing else"?

File: README.md [hunk 15]
```diff
@@ -376,6 +432,22 @@ encoded BibTeX file (with the citation keys serving as local Source IDs). The fi
...
+file, adding `.zip` as filename extension. The filename references in the metadata MUST be kept unchanged.
+
+CLDF processing software MAY implement zip-file discovery, i.e. if a filename referenced in the metadata cannot
+be found, but a file `filename.zip` is found, processing MUST proceed with the unzipped content of `filename.zip`.
+
+If CLDF processing software does not support zip-file discovery, it should signal the corresponding error
```
Comment (chrzyki) [inline README.md:447]: "ZIP file discovery"

File: components/README.md [hunk 16]
```diff
@@ -0,0 +1,5 @@
...
+
+More complex, cross-linguistic datatypes, such as dictionariescan be modeled as [CLDF modules](../modules).
```
Comment (chrzyki) [inline components/README.md:5]: "such as dictionaries, can be ..."

File: components/languages/README.md [hunk 17]
```diff
@@ -11,13 +11,39 @@ Rows in the language table can then be referenced from other tables using a
 
 ## What's a language?
 
-For the purposes of CLDF, a language is one of the objects whose attributes are studied, and stored
-as values for [parameters](../parameters). Thus, if a dataset contains the results of a sociolinguistic
-study, rows in `LanguageTable` will be likely be sociolects. Whether and how such rows are related to
-languoids as understood for example by Glottolog can be indicated in the [Glottocode](http://cldf.clld.org/v1.0/terms.rdf#glottocode) column.
+For the purposes of CLDF, a language is one of the objects whose attributes are described in a dataset.
```
Comment (chrzyki) [inline components/languages/README.md:14]: "For the purposes of CLDF, a language is one of the objects described in a dataset, along with its attributes."

File: components/languages/README.md [hunk 18]
```diff
@@ -11,13 +11,39 @@ Rows in the language table can then be referenced from other tables using a
...
+
+## Speaker area
+
+Datasets may provide information about speaker areas of the languages they describe, i.e. the geographic
+areas where the speakers of the respective languages live.
+
+In this case, `LanguageTable` SHOULD contain a [`speakerArea`](http://cldf.clld.org/v1.0/terms.rdf#speakerArea)
+property, pointing to a media resource in `MediaTable`.
+The linked media resource may be an image of a map, depicting the area, or some other multimedia content 
+for human consumption. But it may also be a [GeoJSON](https://datatracker.ietf.org/doc/html/rfc7946) resource
```
Comment (chrzyki) [inline components/languages/README.md:37]: Remove "but" or replace with "However, ..."?

File: components/examples/description.md (hunk 6 above)
Comment (xrotwang) [reply components/examples/description.md:44]: It actually is copied from `description.md` to `README.md` when the spec is "built". I wanted to make sure that the table of columns for a component is always in sync with the ontology. So, this table is appended to the content of `description.md` and put into `README.md` - which has this special role in GitHub.

File: modules/README.md [hunk 19]
```diff
@@ -0,0 +1,12 @@
+# CLDF Modules
+
+CLDF Modules are used to model complex cross-linguistic datatypes, which typically comprise
+multiple [components](../components) - and may give the contained components particular semantics.
```
Comment (chrzyki) [inline modules/README.md:4]: "and may give the components contained within"?

File: modules/TextCorpus/README.md [hunk 20]
```diff
@@ -0,0 +1,41 @@
+# Text Corpus
+
+Texts can roughly be defined as cohesive stretches of discourse in the object language. They are
+part of the Boasian triad, and fundamental to corpus-based language description.
```
Comment (chrzyki) [inline modules/TextCorpus/README.md:4]: Remove comma.

File: modules/TextCorpus/README.md [hunk 21]
```diff
@@ -0,0 +1,41 @@
...
+In CLDF text corpora are marked by a common property `dc:conformsTo` with value
+[`http://cldf.clld.org/v1.0/terms.rdf#TextCorpus`](http://cldf.clld.org/v1.0/terms.rdf#TextCorpus)
+on a `TableGroup`.
+
+Text Corpora require only an [`ExampleTable`](../../components/examples), listing the lines of text.
+
+
+## Multiple texts
+
+But typically a corpus will contain multiple texts. In this case, metadata about the texts should
```
Comment (chrzyki) [inline modules/TextCorpus/README.md:15]: "Typically, a corpus will contain multiple texts."

File: components/parameternetworks/README.md (hunk 7 above)
Comment (xrotwang) [reply components/parameternetworks/README.md:26]: Do you think this should be explicitly mentioned? I'd include this if there were examples of "Parameter trees" around, but I cannot think of one.

File: terms.rdf [hunk 22]
```diff
@@ -82,6 +82,24 @@
...
     </rdf:Property>
 
+    <rdf:Property rdf:about="http://cldf.clld.org/v1.0/terms.rdf#position" dc:type="generic-property">
+        <rdfs:comment rdf:parseType="Literal">
+            <p>
+                A position of an item in a series or sequence of items. While integer is recommended
+                as datatype, any datatype that can be totally ordered (and this order is transparent
+                e.g. alphabetic order for strings) can be used. A column for this property may also
+                be list-valued (e.g. to implement multi-level orderings), in which case the typical
+                order for tuples is assumed.
```
Comment (chrzyki) [inline terms.rdf:92]: Proposal for reformulating this as I had to read it multiple times. Feel free to ignore:

"A position represents the placement of an item in a series or sequence of items. Although an integer is the recommended datatype, any datatype that supports a total ordering (where the order is transparent, such as alphabetic order for strings) is acceptable. It is also possible to have a list-valued column for this property, which can be useful for implementing multi-level orderings. In such cases, the typical order for tuples is assumed."

File: terms.rdf [hunk 23]
```diff
@@ -240,6 +286,20 @@
         <rdfs:seeAlso rdf:resource="https://en.wikipedia.org/wiki/Foreign_key" />
     </rdf:Property>
 
+    <rdf:Property rdf:about="http://cldf.clld.org/v1.0/terms.rdf#treeReference"
+                  dc:type="reference-property">
+        <rdfs:comment rdf:parseType="Literal">
+            <p>
+                An identifier referencing a language tree by providing a foreign key into TreeTable
```
Comment (chrzyki) [inline terms.rdf:293]: "An identifier referencing a language tree by providing a foreign key **to** TreeTable." - different preposition? At any rate, '.' at the end is missing.

File: terms.rdf [hunk 24]
```diff
@@ -255,6 +315,34 @@
         <dc:hasVersion rdf:resource="http://cldf.clld.org/v1.1"/>
     </rdf:Property>
 
+    <rdf:Property rdf:about="http://cldf.clld.org/v1.0/terms.rdf#speakerArea"
+                  dc:type="reference-property">
+        <rdfs:comment rdf:parseType="Literal">
+            <p>
+                An identifier referencing a media resource by providing a foreign key into MediaTable.
```
Comment (chrzyki) [inline terms.rdf:322]: Again, not sure about the preposition. What I've written above might just be a wrong suggestion, so feel free to ignore.

File: terms.rdf [hunk 25]
```diff
@@ -255,6 +315,34 @@
...
+            </p>
+            <p>
+                This property can be used in <code>LanguageTable</code> to point to a media resource describing
+                the speaker area of a language, i.e. the geographic area where the speakers of the
+                language live.
+            </p>
+            <p>
+                The linked media resource may be an image of a map, depicting the area, or some other
+                multimedia content for human consumption. But it may also be a GeoJSON
+                resource (i.e. a media resource with <code>mediaType</code> <code>application/geo+json</code> ).
```
Comment (chrzyki) [inline terms.rdf:332]: `<code>application/geo+json</code> ).` Remove space.

**Comment by fmatter** [comment]
👍

**Pull request merged**